mathics/doc/common_doc.py

```diff
@@ -1102,6 +1139,10 @@ def doc_part(self, title, modules, builtins_by_module, start):
                         continue
 
                     submodule_name = get_doc_name_from_module(submodule)
+                    # This has the side effect that Symbols with the same
+                    # short name but in different contexts be skipped.
+                    # This happens with ``PlaintextImport`` that appears in
+                    # the HTML and XML contexts.
```

Comment:
The best thing would be to use the part name, chapter name, section name, and submodule name as the key.